### ISSUE TITLE
Deprecation warning from Node

### ISSUE THREAD

**Issue by hjpsievert** [Microsoft/vscode-react-native#398]
## Actual Behavior
1. (node:16676) DeprecationWarning: 'GLOBAL' is deprecated, use 'global'
2. 
debugger worker shows:
// Initialize some variables before react-native code would access them
// and also avoid Node's GLOBAL deprecation warning
var onmessage=null, self=global.GLOBAL=global;
// Cache Node's original require as __debug__.require

## Expected Behavior
1. no deprecation warning
2. 
I assume that should be:
// Initialize some variables before react-native code would access them
// and also avoid Node's GLOBAL deprecation warning
var onmessage=null, self=global, GLOBAL=global;
// Cache Node's original require as __debug__.require

## Software versions
- React-Native VS Code extension version: 0.3.1
- VSCode version: 1.9.1
- OS platform and version: Microsoft Windows 10 Pro 10.0.14393 Build 14393
- NodeJS version: 6.8.0
- React Native version: 0.41.1

## Outputs (Include if relevant)
- Output of the Debug Console (View -> Toggle Debug Console):
```
[vscode-react-native] Executing command: adb -s 169.254.56.136:5555 shell am broadcast -a "com.pdpapptest.RELOAD_APP_ACTION" --ez jsproxy true
[vscode-react-native] Broadcasting: Intent { act=com.pdpapptest.RELOAD_APP_ACTION (has extras) }

Broadcast completed: result=0

[vscode-react-native] Finished executing: adb -s 169.254.56.136:5555 shell am broadcast -a "com.pdpapptest.RELOAD_APP_ACTION" --ez jsproxy true
[vscode-react-native] Starting debugger app worker.
[vscode-react-native] Established a connection with the Proxy (Packager) to the React Native application
[vscode-react-native] Debugger worker loaded runtime on port 27834
(node:4328) DeprecationWarning: 'GLOBAL' is deprecated, use 'global'
[vscode-react-native] Debugger worker loaded runtime on port 40357
(node:16676) DeprecationWarning: 'GLOBAL' is deprecated, use 'global'
Running application "PDPAppTest" with appParams: {"initialProps":{},"rootTag":1}. __DEV__ === true, development-level warning are ON, performance optimizations are OFF

- Output of the React-Native output channel (View -> Toggle Output -> Select React-Native in ListBox):
```
Nothing relevant shown here
```
- Output of the Developer Tools console (Help -> Toggle Developer Tools -> Select Console tab):
```
Nothing that looks relevant
```

**Comment by vladimir-kotikov** [comment]
Hey @hjpsievert, thanks for reporting!

Unfortunately I don't think we can avoid this. As I noted in #394:

> I'm still seeing Node's GLOBAL deprecation message, but I doubt we could avoid it because GLOBAL's setter is also configured to emit this warning (see https://github.com/nodejs/node/blob/master/lib/internal/bootstrap_node.js#L203-L219)

So this warning will be emitted at least once, when we try to write `GLOBAL` property

As for the change you suggested, I don't think it would help since you are basically proposing to define _module local_ variable named `GLOBAL`, however the intent was to overwrite 'global' `GLOBAL`. Moreover in this case, since deprecation warning is not triggered early before react native code is loaded, it will be triggered after then by react - in my case it leads to the red box shown on the screen.

**Comment by hjpsievert** [comment]
Hello Vladimir:

Thanks for the update and detail. So does that mean that the newest version of react-native tools on Visual Studio Code is just fine and these are erroneous deprecation warnings that I can ignore or is there a future change that will make them go away.

 

I have also been seeing error messages about promises and timeouts after setting breakpoints that started a few days ago when I updated react-native tools, but breakpoints work just fine. Is that something I should report the next time I see them or is that another known ‘fake’ problem?

 

 

Best regards,

Hans-Jürgen (HJ) Sievert
:: 3770 Vancouver Drive, Reno, NV 89511
:: Cell (775) 742-7425 Home (775) 853-4599

::  <mailto:hj.sievert@gmail.com> hj.sievert@gmail.com

:: http://www.linkedin.com/in/hjsievert

**Comment by vladimir-kotikov** [comment]
> So does that mean that the newest version of react-native tools on Visual Studio Code is just fine and these are erroneous deprecation warnings that I can ignore or is there a future change that will make them go away.

Yes, that's absolutely fine, though it might be annoying. I guess it will go away once Node will completely remove `global.GLOBAL` (see also https://github.com/nodejs/node/wiki/Breaking-changes-between-v5-and-v6#globals)

> I have also been seeing error messages about promises and timeouts after setting breakpoints that started a few days ago when I updated react-native tools, but breakpoints work just fine.

No, this is somenting we haven't seen before. Feel free to open a new issue when they appear next time.

**Comment by jhalborg** [comment]
Hey :) 

I'm getting this error, and it's crashing my app when continuing from the last breakpoint. 

Using vscode 1.15.1
vscode-react-native: 0.4.1
Expo: 20.0

Any way to suppress the warning?

**Comment by vladimir-kotikov** [comment]
@jhalborg those warnings shoud have went away after #401, so this might be a different issue. Does this reproduce on clean Expo project? Mac OS/Windows? Which Node JS version are you running?

**Comment by vladimir-kotikov** [comment]
@jhalborg Ideally I'd appreciate if you could create a separate issue and file all details in issue template. Thanks in advance.
